src/index.ts

```diff
@@ -94,6 +109,44 @@ function getLoaderOptions(loader: interfaces.Webpack) {
 }
 
 /**
+ * Get any files generated by previous loaders in the same chain.
+ */
+function getGeneratedFiles(loader: interfaces.Webpack) {
+    const generatedFiles: { [filePath: string]: string } = {};
+    for (let i = loader.loaders.length - 1; i > loader.loaderIndex; i--) {
+        const data = loader.loaders[i].data;
+        const files = data && data['ts-loader-files'];
+        if (files) {
+            for (let filePath of Object.keys(files)) {
+                if (filePath) {
+                    generatedFiles[path.normalize(filePath)] = files[filePath];
+                }
+            }
+        }
+    }
+    return generatedFiles;
+}
+
+/**
+ * Remove dependencies specified by argument exclude.
+ */
+function removeDependencies(loader: interfaces.Webpack, exclude: (filePath: string) => boolean): void {
```

Comment:
I'm basically happy with this - I'd just like to do some variable renames to improve clarity.  Please could

- `dependencies` become `dependenciesToPreserve`
- `contextDependencies` become `contextDependenciesToPreserve`

Bit of a mouthful but it makes intent obvious